pkg/importer/dataStream_test.go

```diff
@@ -341,353 +307,67 @@ func TestCopyImage(t *testing.T) {
-	}{
-	// TODO: Add test cases.
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			d := &dataStream{
-				Url:         tt.fields.Url,
-				Readers:     tt.fields.Readers,
-				buf:         tt.fields.buf,
-				qemu:        tt.fields.qemu,
-				Size:        tt.fields.Size,
-				accessKeyId: tt.fields.accessKeyId,
-				secretKey:   tt.fields.secretKey,
-			}
-			if err := d.constructReaders(); (err != nil) != tt.wantErr {
-				t.Errorf("dataStream.constructReaders() error = %v, wantErr %v", err, tt.wantErr)
-			}
-		})
-	}
-}
-
-func Test_dataStream_topReader(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
-	tests := []struct {
-		name   string
-		fields fields
-		want   io.ReadCloser
-	}{
-	// TODO: Add test cases.
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			d := dataStream{
-				Url:         tt.fields.Url,
-				Readers:     tt.fields.Readers,
-				buf:         tt.fields.buf,
-				qemu:        tt.fields.qemu,
-				Size:        tt.fields.Size,
-				accessKeyId: tt.fields.accessKeyId,
-				secretKey:   tt.fields.secretKey,
-			}
-			if got := d.topReader(); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("dataStream.topReader() = %v, want %v", got, tt.want)
-			}
-		})
-	}
-}
-
-func Test_dataStream_fileFormatSelector(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
-	type args struct {
-		hdr *image.Header
-	}
-	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		wantErr bool
-	}{
-	// TODO: Add test cases.
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			d := &dataStream{
-				Url:         tt.fields.Url,
-				Readers:     tt.fields.Readers,
-				buf:         tt.fields.buf,
-				qemu:        tt.fields.qemu,
-				Size:        tt.fields.Size,
-				accessKeyId: tt.fields.accessKeyId,
-				secretKey:   tt.fields.secretKey,
-			}
-			if err := d.fileFormatSelector(tt.args.hdr); (err != nil) != tt.wantErr {
-				t.Errorf("dataStream.fileFormatSelector() error = %v, wantErr %v", err, tt.wantErr)
-			}
-		})
-	}
-}
+	imageDir, _ := filepath.Abs("../../test/images")
+	localImageBase := filepath.Join("file://", imageDir)
 
-func Test_dataStream_gzReader(t *testing.T) {
-	type fields struct {
-		Url         *url.URL
-		Readers     []Reader
-		buf         []byte
-		qemu        bool
-		Size        int64
-		accessKeyId string
-		secretKey   string
-	}
 	tests := []struct {
 		name    string
-		fields  fields
-		want    *Reader
-		want1   int64
+		ds      *dataStream
 		wantErr bool
 	}{
-	// TODO: Add test cases.
+		{
+			name:    "successfully construct tar.gz reader",
+			ds:      createSimpleDataStream(filepath.Join(localImageBase, "sample.tar.gz"), "", ""),
```

Comment:
May save some pain down the line to globally constify all known file names
`sample.tar.gz`
`sample.tar.xz`
`tinyCore.iso`
`tinyCorebad.iso`